homeassistant/components/image_processing/facebox.py

```diff
@@ -75,15 +80,40 @@ def parse_faces(api_faces):
     return known_faces
 
 
-def post_image(url, image):
+def post_image(url, username, password, image):
     """Post an image to the classifier."""
     try:
         response = requests.post(
             url,
+            auth=requests.auth.HTTPBasicAuth(username, password),
             json={"base64": encode_image(image)},
             timeout=TIMEOUT
             )
-        return response
+        if response.status_code == HTTP_UNAUTHORIZED:
+            _LOGGER.error("AuthenticationError on %s", CLASSIFIER)
+            return None
+        else:
+            return response
+    except requests.exceptions.ConnectionError:
+        _LOGGER.error("ConnectionError: Is %s running?", CLASSIFIER)
+
+
+def teach_file(url, username, password, name, file_path):
+    """Teach the classifier a name associated with a file."""
+    try:
+        with open(file_path, 'rb') as open_file:
+            response = requests.post(
+                url,
+                auth=requests.auth.HTTPBasicAuth(username, password),
+                data={FACEBOX_NAME: name, ATTR_ID: file_path},
+                files={'file': open_file},
+                timeout=TIMEOUT
```

Comment:
Be careful with the default timeout as this is sending files

Reply:
Will remove